src/Storages/HDFS/HDFSCommon.cpp

```diff
@@ -0,0 +1,183 @@
+#include <Storages/HDFS/HDFSCommon.h>
+#include <Poco/URI.h>
+#include <boost/algorithm/string/replace.hpp>
+
+#if USE_HDFS
+#include <Common/ShellCommand.h>
+#include <Common/Exception.h>
+#include <IO/WriteBufferFromString.h>
+#include <IO/Operators.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int BAD_ARGUMENTS;
+extern const int NETWORK_ERROR;
+extern const int EXCESSIVE_ELEMENT_IN_CONFIG;
+}
+
+const String HDFSBuilderWrapper::CONFIG_PREFIX = "hdfs";
+
+void HDFSBuilderWrapper::loadFromConfig(const Poco::Util::AbstractConfiguration & config,
+    const String & config_path, bool isUser)
+{
+    Poco::Util::AbstractConfiguration::Keys keys;
+
+    config.keys(config_path, keys);
+    for (const auto & key : keys)
+    {
+        const String key_path = config_path + "." + key;
+
+        String key_name;
+        if (key == "hadoop_kerberos_keytab")
+        {
+            needKinit = true;
+            hadoop_kerberos_keytab = config.getString(key_path);
+            continue;
+        }
+        else if (key == "hadoop_kerberos_principal")
+        {
+            needKinit = true;
+            hadoop_kerberos_principal = config.getString(key_path);
+
+#if USE_INTERNAL_HDFS3_LIBRARY
+            hdfsBuilderSetPrincipal(hdfs_builder, hadoop_kerberos_principal.c_str());
+#endif
+
+            continue;
+        }
+        else if (key == "hadoop_kerberos_kinit_command")
+        {
+            needKinit = true;
+            hadoop_kerberos_kinit_command = config.getString(key_path);
+            continue;
+        }
+        else if (key == "hadoop_security_kerberos_ticket_cache_path")
+        {
+            if (isUser)
+            {
+                throw Exception("hadoop.security.kerberos.ticket.cache.path cannot be set per user",
+                    ErrorCodes::EXCESSIVE_ELEMENT_IN_CONFIG);
+            }
+
+            hadoop_security_kerberos_ticket_cache_path = config.getString(key_path);
+            // standard param - pass further
+        }
+
+        key_name = boost::replace_all_copy(key, "_", ".");
+
+        const auto & [k,v] = keep(key_name, config.getString(key_path));
+        hdfsBuilderConfSetStr(hdfs_builder, k.c_str(), v.c_str());
+    }
+}
+
+String HDFSBuilderWrapper::getKinitCmd()
+{
+    WriteBufferFromOwnString ss;
+
+    String cache_name =  hadoop_security_kerberos_ticket_cache_path.empty() ?
+        String() :
+        (String(" -c \"") + hadoop_security_kerberos_ticket_cache_path + "\"");
+
+    ss << hadoop_kerberos_kinit_command << cache_name <<
```

Comment:
Add a comment with a command example?

Reply:
Ok, I will.
And I'll add something to the doc, while the configuration parameter is listed, while not documented at all.